internal/shim-sgx/src/handler/enarx.rs

```diff
@@ -1,9 +1,35 @@
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::sgx_uarch::REPORT;
+use crate::sgx_uarch::TARGETINFO;
+use core::arch::asm;
 use sallyport::syscall::{BaseSyscallHandler, EnarxSyscallHandler, SGX_QUOTE_SIZE, SGX_TECH};
 use sallyport::untrusted::{UntrustedRef, UntrustedRefMut, ValidateSlice};
 use sallyport::Reply;
 
+/// Invoke EREPORT
+#[allow(dead_code)]
+fn enclu_ereport(targetinfo: &TARGETINFO, reportdata: &[u8; 64]) -> crate::sgx_uarch::REPORT {
+    const EREPORT: usize = 0;
+
+    let mut report = core::mem::MaybeUninit::<REPORT>::uninit();
+
+    unsafe {
+        asm!(
+            "xchg       {RBX}, rbx",
+            "enclu",
+            "mov        rbx, {RBX}",
+
+            RBX = inout(reg) targetinfo => _,
+            in("rax") EREPORT,
+            in("rcx") reportdata.as_ptr(),
+            in("rdx") report.as_mut_ptr(),
+        );
+    }
+
+    unsafe { report.assume_init() }
+}
+
```

Comment:
Right, that will also remove the need for `dead_code`. It's cool.

Reply:
I'll still keep this instead of old `get_report`. It's not as fancy but it's very documentation, easy to trace (and grep) function name. You know exactly where it is coming from with this name.